diff --git a/node_modules/@cesium/engine/Source/Scene/Scene.js b/node_modules/@cesium/engine/Source/Scene/Scene.js
index 127c6aa..1ef37ba 100644
--- a/node_modules/@cesium/engine/Source/Scene/Scene.js
+++ b/node_modules/@cesium/engine/Source/Scene/Scene.js
@@ -2404,6 +2404,7 @@ function executeCommands(scene, passState) {
       }
     }
 
+    render_external_(scene, frustum, "GLOBE");
     const globeDepth = view.globeDepth;
     if (defined(globeDepth) && environmentState.useGlobeDepthFramebuffer) {
       globeDepth.executeCopyDepth(context, passState);
@@ -2584,6 +2585,10 @@ function executeCommands(scene, passState) {
       executeCommand(commands[j], scene, context, passState);
     }
 
+    render_external_(scene, frustum, "ENTITY");
+
+    render_external_(scene, frustum, "SCREEN");
+
     if (index !== 0 && scene.mode !== SceneMode.SCENE2D) {
       // Do not overlap frustums in the translucent pass to avoid blending artifacts
       frustum.near = frustumCommands.near;
@@ -3691,7 +3696,14 @@ function postPassesUpdate(scene) {
 }
 
 const scratchBackgroundColor = new Color();
-
+function render_external_(scene, frustum, pass) {
+  if (scene.render_external_frame_functions) {
+    var functions = scene.render_external_frame_functions;
+    for (var i = 0; i < functions.length; i++) {
+      functions[i](scene, frustum, pass);
+    }
+  }
+}
 function render(scene) {
   const frameState = scene._frameState;
 
